### PR TITLE
build: update Bazel dependency to 0.11.1

### PR DESCRIPTION
**Pull request by alexeagle**
This fixes some minor issues we've seen, and ensures that we report issues against the current release.

### CODE REVIEW

**Comment by alexeagle** [comment]
Note, this PR upgrades Skylint and includes new deprecations, however due to https://github.com/bazelbuild/bazel/issues/4858 we cannot fix the call to `ctx.new_file(genfiles_dir)` so I turned off that lint check altogether.
/cc @laurentlb

File: .gitignore [hunk 1]
```diff
@@ -14,6 +14,7 @@ pubspec.lock
 .settings/
 *.swo
 modules/.settings
+.bazelrc
```
Comment (IgorMinar) [inline .gitignore]: is this right? why ignore it?

**Review by IgorMinar — APPROVED** [review]

**Comment by IgorMinar** [comment]
http://test/OCL:192401172:BASE:192401438:1523422636784:b4aaf822

File: .gitignore (hunk 1 above)
Comment (alexeagle) [reply .gitignore]: I've set up build event logging to cloud, and I put my local settings in this file. Bazel documentation recommends using this file for non-shared settings that are project-specific.

**Pull request closed without merging**